labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -28,40 +28,61 @@ object RecursionPatternMatchingExercise {
    * checkValuesIncrease(Seq(1,2,3)) == true
    * checkValuesIncrease(Seq(1,2,2)) == false
    */
-  def checkValuesIncrease(seq: Seq[Int]): Boolean = {
-    error("fix me")
+  def checkValuesIncrease(seq: Seq[Int]): Boolean = seq match{
+    case Nil => true
+    case _ => true
+    case left :: right :: rest => left > right && checkValuesIncrease(rest)
+    
   }
   
   /**
    * Group Consecutive values
    * List(1,1,2,3,1,1) -> List(1,1), List(2), List(3), List(1,1)
    */
-  def groupConsecutive[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+  def groupConsecutive[T](in: List[T]): List[List[T]] = in match {
+    case Nil => Nil
+    case value :: junk => {
+      val (currentValues, rest) = in.span(_ == value)
+      currentValues :: groupConsecutive(rest)
+    }
   }
 
   /**
    * Group Equal values
    * List(1,1,2,3,1,1) -> List(1,1,1,1), List(2), List(3)
    */
-  def groupEquals[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+  def groupEquals[T](in: List[T]): List[List[T]] = in match {
+    case Nil => Nil
+    case value :: junk => {
+      val (currentValues, rest) = in.partition(_ == value)
+      currentValues :: groupEquals(rest)
+    }
+    
   }
 
   /**
    * Compress values
    * List(1,1,2,3,1,1) -> List(1,2,3)
    */
-  def compress[T](in: List[T]): List[T] = {
-    error("fix me")
+  def compress[T](in: List[T]): List[T] = in match {
+    case Nil => Nil
+    case value :: junk => {
+      val (currentValues, rest) = in.span(_ == value)
+      value :: compress(rest)
+    }
   }
   
   /**
    * Define the amount of all equal members
    * List(1,1,2,3,1,1) -> List((4,1),(1,2),(1,3))
    */
-  def amountEqualMembers[T](in: List[T]): List[(Int, T)] = {
-    error("fix me")
+  def amountEqualMembers[T](in: List[T]): List[(Int, T)] = in match {
+    case Nil => Nil
+    case value :: junk => {
+      val (currentValues, rest) = in.span(_ == value)
+      val numVal = currentValues.length
+      (numVal, value) :: amountEqualMembers(rest)
+    }
```

Comment:
Same as above - I think that span is too local (you need to count *all* the 1's).